code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -293,6 +293,9 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 	if(!sharer)
 		return 0
 
+	archive()

```

Comment:
This only needs to be done for the non-turf mixtures, there's no need to update in usual use case. I'd rather not cause needless updates with every share call. Wrapper or parameter would be good here.